components/script/dom/htmlmediaelement.rs

```diff
@@ -508,22 +521,41 @@ impl HTMLMediaElement {
             let (action_sender, action_receiver) = ipc::channel().unwrap();
             let window = window_from_node(self);
             let script_chan = window.networking_task_source();
-            let listener = box NetworkListener {
+            let listener = NetworkListener {
                 context: context,
                 script_chan: script_chan,
                 wrapper: Some(window.get_runnable_wrapper()),
             };
 
-            let response_target = AsyncResponseTarget {
-                sender: action_sender,
-            };
             ROUTER.add_route(action_receiver.to_opaque(), box move |message| {
-                listener.notify_action(message.to().unwrap());
+                listener.notify_fetch(message.to().unwrap());
             });
 
             // FIXME: we're supposed to block the load event much earlier than now
-            let doc = document_from_node(self);
-            doc.load_async(LoadType::Media(url), response_target, None);
+            let document = document_from_node(self);
+
+            let ty = if self.is::<HTMLAudioElement>() {
+                RequestType::Audio
+            } else if self.is::<HTMLVideoElement>() {
+                RequestType::Video
+            } else {
+                unreachable!("Unexpected HTMLMediaElement")
+            };
+
+            let request = RequestInit {
```

Comment:
Yes, it is; specialized to the case where we don't pay attention to the crossorigin attribute.